Formula/dafny.rb

```diff
@@ -0,0 +1,47 @@
+class Dafny < Formula
+  desc "Verification-aware programming language"
+  homepage "https://www.microsoft.com/en-us/research/project/dafny-a-language-and-program-verifier-for-functional-correctness"
+  url "https://github.com/dafny-lang/dafny/archive/v2.3.0.tar.gz"
+  sha256 "ea7ae310282c922772a46a9a85e2b4213043283038b74d012047b5294687d168"
+
+  depends_on "mono"
+  depends_on "mono-libgdiplus" => :build
+  depends_on "z3"
+
+  resource "boogie" do
+    url "https://github.com/boogie-org/boogie.git",
```

Comment:
@Homebrew/core not entirely sure on the rules for resources. Does this need a tagged release too?

Reply:
There is no tagged release available for this. I can only specify the revision, and I saw some formulas did the same thing. Do you have a better solution?

Reply:
We don't need a tagged release but we do need a specific revision.

Reply:
@MikeMcQuaid I had specified the revision. Do I need to make any change on this?

Reply:
I didn't see that, sorry. No, all good then.